src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+                )
+                if attempt < max_retries - 1:
+                    continue
+                return False
+
+            with open(src_path, 'wb') as f:
+                f.write(response.content)
+
+            if _unzip_file(src_path, '.'):
+                os.rename(extracted_folder, target_folder)
+                print("Successfully downloaded aws-greengrass-lite")
+                return True
+
+        except requests.exceptions.RequestException as e:
+            print(f"Error when downloading aws-greengrass-lite, {str(e)}")
+            if attemp == max_retries - 1:
+                print(
+                    f"Failed to download aws-greengrass-lite after {max_retries} attempts"
+                )
+                return False
+
+
+def _unzip_file(src_path: str, dest_path: str) -> bool:
+    try:
+        with zipfile.ZipFile(src_path, 'r') as zip_ref:
+            zip_ref.extractall(dest_path)
+
+        os.remove(src_path)
+        return True
+
+    except Exception as e:
+        print(f"Error during extraction: {str(e)}")
+        return False
+
+
+def _install_build_dependencies() -> bool:
+    try:
+        # List of packages to install
+        packages = [
+            "build-essential", "pkg-config", "cmake", "git", "curl",
+            "libssl-dev", "libcurl4-openssl-dev", "uuid-dev", "libzip-dev",
+            "libsqlite3-dev", "libyaml-dev", "libsystemd-dev", "libevent-dev",
+            "liburiparser-dev", "cgroup-tools"
+        ]
+
+        # Update package list
+        update_command = ["sudo", "apt", "update"]
+        subprocess.run(update_command, check=True)
+
+        # Install packages
+        install_command = ["sudo", "apt", "install", "-y"] + packages
+        subprocess.run(install_command, check=True)
+
+        print("Successfully updated and installed dependencies")
+        return True
+
+    except Exception as e:
+        print(f"Error when installing dependencies: {e}", file=sys.stderr)
+        return False
+
+
+def _add_user_and_group(user: str, group: str):
+    try:
+        # Create the group
+        if not _check_group_exists(group):
+            subprocess.run(['sudo', 'groupadd', group], check=True)
+            print(f"Successfully created {group} group")
+
+        # Create the user and add to the group
+        if not _check_user_exists(user):
+            subprocess.run(['sudo', 'useradd', '-Ng', group, user], check=True)
+            print(f"Successfully created {user} user")
+        return True
+
+    except subprocess.CalledProcessError as e:
+        print(f"Command failed: {e.cmd}. Return code: {e.returncode}")
+        return False
+    except Exception as e:
+        print(f"Error when adding user and group: {str(e)}")
+        return False
+
+
+def _delete_user_and_group(user: str, group: str):
+    try:
+        # Delete the user
+        if _check_user_exists(user):
+            subprocess.run(['sudo', 'userdel', user], check=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method is running with elevated privileges. Privilege elevation occurs when you access resources with an extended set of permissions. When unattended, these privileges might allow attackers to compromise the system. Do not use elevated privileges when they can be avoided.

[Learn more](https://cwe.mitre.org/data/definitions/269.html)

Similar issue at line number 259.